src/Components/Facility/LocationManagement.tsx

```diff
@@ -7,20 +7,51 @@ import routes from "../../Redux/api";
 import PaginatedList from "../../CAREUI/misc/PaginatedList";
 import { LocationModel } from "./models";
 import RecordMeta from "../../CAREUI/display/RecordMeta";
+import request from "../../Utils/request/request";
+import * as Notification from "../../Utils/Notifications.js";
+import ConfirmDialog from "../Common/ConfirmDialog";
 
 const Loading = lazy(() => import("../Common/Loading"));
 
 interface Props {
   facilityId: string;
 }
 
+interface LocationProps extends LocationModel {
+  setShowDeletePopup: any;
+}
+
 export default function LocationManagement({ facilityId }: Props) {
+  const [showDeletePopup, setShowDeletePopup] = useState({
+    open: false,
+    name: "",
+    id: "",
+  });
+
+  const deleteAssetLocation = async () => {
+    const res: any = await request(routes.deleteFacilityAssetLocation, {
+      pathParams: {
+        facility_external_id: facilityId,
+        external_id: showDeletePopup.id,
+      },
+    });
+    if (res?.res?.status === 204) {
+      Notification.Success({
+        msg: `Location ${showDeletePopup.name} deleted successully`,
+      });
+      PaginatedList.Refresh;
+    } else {
+      Notification.Error({ msg: "Something went wrong!" });
+    }
+    setShowDeletePopup({ ...showDeletePopup, open: false });
+  };
```

Comment:
1. Avoid using `any`. `useQuery` and `request` is typesafe.
2. You need not dispatch the error notification by yourself. `useQuery`/`request` will do it for you.
3. What is `PaginatedList.Refresh` doing here?

```suggestion
    const { res } = await request(routes.deleteFacilityAssetLocation, {
      pathParams: {
        facility_external_id: facilityId,
        external_id: showDeletePopup.id,
      },
    });
    if (res?.ok) {
      Notification.Success({
        msg: `Location ${showDeletePopup.name} deleted successully`,
      });
    }
    setShowDeletePopup({ ...showDeletePopup, open: false });
  };
```